### ISSUE TITLE
Multiple Set-Cookie headers get aggregated

### ISSUE THREAD

**Issue by Michael-Dawkins** [stoplightio/prism#1190]
**Describe the bug**
HTTP response that set multiple cookies that pass through prism proxy see their set-cookie headers merged, breaking the client understanding of them.

**To Reproduce**

1. Make a web server that sets 2 or more cookies, (Nodejs and Java servlet both set them as separate Set-Cookie headers in the same HTTP response)
2. fetch the server through prism proxy on a web page
3. The HTTP response obtained by the browser (or POSTMAN shows it aswell) contains only one Set-Cookie, with the two cookies aggregated and separated by a space
Both Chrome and Firefox only aknoledge the first cookie and ignore the second, which, in my case, breaks authentication.

**Expected behavior**
I expect the multiple Set-Cookie headers to be preserved.

**Environment (remove any that are not applicable):**
 - Library version: 3.3.4
 - OS: Windows and MacOS
 - Browser: Chrome 83 and Firefox

**Comment by philsturgeon** [comment]
Could I ask, what API authentication requires cookies?

**Comment by Michael-Dawkins** [comment]
It is our own APIs, called by the frontend. Just curious, is it a surprising use case?
By using cookies with "HTTP ONLY", we can make sure the frontend has no read access to the tokens, to protect against XSS attacks.
But I am open to suggestions.

**Comment by philsturgeon** [comment]
It's not hugely common, but it is done. 

https://developer.atlassian.com/cloud/jira/platform/jira-rest-api-cookie-based-authentication/

We should not be combining cookies regardless, prism shouldn't mess with headers for no reason.

@Michael-Dawkins could you give us an example OpenAPI and curl request so that we can clearly see exactly what is happening? Reproducible bugs are solved a lot quicker. :)

**Comment by Lidbetter** [comment]
Hi, I am also running into this problem, here are the 2 sets of raw headers returned (according to firefox) from my backend directly and through prism proxy (docker stoplight/prism:4.1.0)

Direct (works as expected):
```
HTTP/1.1 200 OK
Server: nginx/1.17.0
Content-Type: application/json;charset=utf-8
Transfer-Encoding: chunked
Connection: keep-alive
Set-Cookie: app2=432dfe8a3addeb0cbbdca5d65437aeb9; path=/; domain=.app2.net; HttpOnly; SameSite=Lax
Expires: Thu, 19 Nov 1981 08:52:00 GMT
Cache-Control: no-store, no-cache, must-revalidate
Pragma: no-cache
Set-Cookie: app2_remember=%3A; expires=Thu, 15-Oct-2020 23:00:56 GMT; Max-Age=0; path=/; domain=.app2.net; HttpOnly; SameSite=lax
Cache-Control: no-cache, private
Date: Fri, 16 Oct 2020 00:00:56 GMT
X-Content-Type-Options: nosniff
X-XSS-Protection: 1; mode=block
```

Via Prism (breaks sessions, invalid set-cookie header):
```
HTTP/1.1 200 OK
server: nginx/1.17.0
content-type: application/json;charset=utf-8
set-cookie: app2=7d77dc3f4475c8a1502164562930f79f; path=/; domain=.app2.net; HttpOnly; SameSite=Lax app2_remember=%3A; expires=Thu, 15-Oct-2020 22:59:10 GMT; Max-Age=0; path=/; domain=.app2.net; HttpOnly; SameSite=lax
expires: Thu, 19 Nov 1981 08:52:00 GMT
cache-control: no-store, no-cache, must-revalidate, no-cache, private
pragma: no-cache
date: Thu, 15 Oct 2020 23:59:11 GMT
vary: Origin
access-control-max-age: 21600
access-control-allow-credentials: true
access-control-allow-origin: http://localdev.app2.net:11180
access-control-allow-methods: OPTIONS, GET, POST, PUT, PATCH, DELETE
access-control-allow-headers: Authorization, Accept, Content-Type, Origin, Set-Cookie, X-Requested-With, X-Auth-Token
x-content-type-options: nosniff
x-xss-protection: 1; mode=block
Content-Length: 460
Connection: keep-alive
```

Because of this conatination firefox is interperting this as a single cookie (`app2`) insetead of 2 (`app2`, `app2_remember`) and then immeditly removing the `app2` cookie from the browser because the expire date of the "remember cookie" is in the past 


As a side note, the `app2_remember` "remember me" cookie is deliberetly being sent with an expire date in the past so that any invalid previous value of that cookie will be automatically cleared.

**Comment by XVincentX** [comment]
@Lidbetter Thanks for the detailed report. We'll try to take a look into this.

**Comment by XVincentX** [comment]
Closed by #1489

**Comment by XVincentX** [comment]
@Michael-Dawkins While we haven't been able to tame this correctly, now we are at least separating multiple headers with commas; while there might still be some edge cases, your use case should be covered.

Please give it a try from the master branch and feel free to reopen in case it's still incorrect.
